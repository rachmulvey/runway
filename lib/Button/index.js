import React from 'react';

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _objectWithoutProperties(source, excluded) {
  if (source == null) return {};

  var target = _objectWithoutPropertiesLoose(source, excluded);

  var key, i;

  if (Object.getOwnPropertySymbols) {
    var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

    for (i = 0; i < sourceSymbolKeys.length; i++) {
      key = sourceSymbolKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
      target[key] = source[key];
    }
  }

  return target;
}

var Button = function Button(_ref) {
  var className = _ref.className,
      label = _ref.label,
      buttonProps = _objectWithoutProperties(_ref, ["className", "label"]);

  return React.createElement("div", {
    className: className
  }, React.createElement("button", buttonProps, label));
};

Button.defaultProps = {
  label: "button"
};

export default Button;
