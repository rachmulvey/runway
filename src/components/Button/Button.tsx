import * as React from 'react';

export interface Props {
  className: string;
  label: number;
}

const Button = ({ className, label, ...buttonProps }: Props) => {
  return (
    <div className={className}>
      <button {...buttonProps}>
        {label}
      </button>
    </div>
  )
}

Button.defaultProps = {
  label: "button"
};

export default Button;
