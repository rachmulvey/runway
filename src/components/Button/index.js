import { css } from 'emotion';
import React from 'react';
import { CSS_SELECTOR_ACTIVE, CSS_SELECTOR_HOVER } from '../../constants/css';
import theme from '../../theme/airways';

export const BUTTON_VARIANT_SOLID = 'solid';
export const BUTTON_VARIANT_HOLLOW = 'hollow';

const baseStyles = css({
  fontFamily: theme.fontFamily.main,
  fontSize: theme.fontSize.button,
  width: theme.buttons.width,
  height: theme.buttons.height,
  borderRadius: theme.buttons.borderRadius,
  borderWidth: 0,
  padding: '14px 24px 15px',
  fontWeight: 700,
  lineHeight: 1.56,
  letterSpacing: '0.1em',
  textTransform: 'uppercase',
  cursor: 'pointer',
  [CSS_SELECTOR_ACTIVE]: {
    transform: `translate(${theme.buttons.clickOffset}, ${
      theme.buttons.clickOffset
    })`
  }
});

const solidStyles = css(
  {
    backgroundColor: theme.colours.primary,
    color: theme.colours.white,
    [CSS_SELECTOR_HOVER]: {
      backgroundColor: theme.colours.primaryDark
    },
    [CSS_SELECTOR_ACTIVE]: {
      backgroundColor: theme.colours.primaryDark
    }
  },
  baseStyles
);

const hollowStyles = css(
  {
    backgroundColor: theme.colours.transparent,
    boxShadow: `0 0 0 2px ${theme.colours.primary} inset`,
    color: theme.colours.primary,
    fontWeight: 'normal',
    [CSS_SELECTOR_HOVER]: {
      boxShadow: `0 0 0 2px ${theme.colours.primaryDark} inset`,
      color: theme.colours.primaryDark
    },
    [CSS_SELECTOR_ACTIVE]: {
      boxShadow: `0 0 0 2px ${theme.colours.primaryDark} inset`,
      color: theme.colours.primaryDark
    }
  },
  baseStyles
);

const styleMap = {
  [BUTTON_VARIANT_SOLID]: solidStyles,
  [BUTTON_VARIANT_HOLLOW]: hollowStyles
};

const Button = ({ className, label, variant, ...buttonProps }) => (
  <div className={className}>
    <button type="button" {...buttonProps} css={styleMap[variant]}>
      {label}
    </button>
  </div>
);

Button.defaultProps = {
  label: 'button',
  variant: BUTTON_VARIANT_SOLID
};

export default Button;
