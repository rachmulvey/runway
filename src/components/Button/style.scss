@import '../../style/_variables.scss';

$width: 100%;
$height: 50px;
$font-size: 16px;
$border-radius: 4px;
$click-offset: 1px;

.button {
  font-family: 'Ciutadella', sans-serif;
  font-size: $font-size;
  width: $width;
  height: $height;
  border-radius: $border-radius;
  padding: 14px 24px 15px;
  font-family: inherit;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.56;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;

  &:active {
    transform: translate($click-offset, $click-offset);
  }
}

.hollow {
  background-color: transparent;
  border-width: 0;
  box-shadow: 0 0 0 2px $color-qantas-red inset;
  color: $color-qantas-red;
  font-weight: normal;

  &:hover {
    box-shadow: 0 0 0 2px $color-dark-red inset;
    color: $color-dark-red;
  }

  &:active {
    box-shadow: 0 0 0 2px $color-focus-red inset;
    color: $color-focus-red;
  }
}

.solid {
  border-width: 0;
  background-color: $color-qantas-red;
  color: $color-white;

  &:hover {
    background-color: $color-dark-red;
  }

  &:active {
    background-color: $color-focus-red;
  }
}
