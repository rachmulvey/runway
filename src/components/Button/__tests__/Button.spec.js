import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../Button';
describe('Button', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Button debug />);

    expect(component).toMatchSnapshot();
  });
});
