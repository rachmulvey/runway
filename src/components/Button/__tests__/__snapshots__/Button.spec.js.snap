// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly in "debug" mode 1`] = `
<div>
  <button
    debug={true}
  >
    button
  </button>
</div>
`;
